### ISSUE TITLE
PostgreSQL 15 - syntax sensitivity

### ISSUE THREAD

**Issue by seven0f** [meeting-room-booking-system/mrbs-code#3526]
**Describe the bug**
After migrating to a new host, site fails to load with "Fatal error: unfortunately the database is not available at the moment." The following error message is logged in php log:

```
[01-Oct-2023 14:21:31 Europe/London] Uncaught exception 'MRBS\DBException' in /home/[redacted]/public_html/lib/MRBS/DB.php at line 228
SQLSTATE[42601]: Syntax error: 7 ERROR:  trailing junk after parameter at or near "$1O"
LINE 4:             WHERE table_name = $1ORDER BY ordinal_position
                                       ^
SQL: SELECT column_name, column_default, data_type, numeric_precision, numeric_scale,                   character_maximum_length, character_octet_length, is_nullable            FROM information_schema.columns            WHERE table_name = ?ORDER BY ordinal_position
Params: Array
(
    [0] => mrbs_room
)

#0 /home/[redacted]/public_html/lib/MRBS/DB_pgsql.php(305): MRBS\DB->query()
#1 /home/[redacted]/public_html/lib/MRBS/DB.php(335): MRBS\DB_pgsql->field_info()
#2 /home/[redacted]/public_html/functions.inc(1489): MRBS\DB->field_exists()
#3 /home/[redacted]/public_html/standard_vars.inc.php(22): MRBS\get_default_area()
#4 /home/[redacted]/public_html/defaultincludes.inc(36): require('...')
#5 /home/[redacted]/public_html/index.php(8): require('...')
#6 {main}

MRBS GET: Array
(
)
MRBS POST: Array
(
)
MRBS SESSION: Array
(
    [last_page] => 
    [this_page] => /
)

```
**To Reproduce**
Migrate system to one running PostgreSQL 15, or upgrade PostgreSQL to version 15.

**Expected behavior**
Site would load and function as normal.

**Server details (please complete the following information):**
 - MRBS version 1.11.0
 - Web server: Apache 2.4.57
 - OS: Debian 12
 - PHP version: 8.2.7
 - Database used: PostgreSQL 15.3 (Debian 15.3-0+deb12u1)

**Additional context**
Coming from PostgreSQL 14.

I can fix this by editing line 296 of file `/home/[redacted]/public_html/lib/MRBS/DB_pgsql.php`:

`WHERE table_name = ?";` to `WHERE table_name = ? ";`

This brings the site back and, with limited testing, it appears to be working corretly. There are a number of other places in the file above where `= ?";` appears, which suggests the problem might resurface elsewhere.

[This post](https://stackoverflow.com/questions/75282073/postgresql-15-syntax-sensitivity) on stackoverflow showed me what to look for.

**Comment by jberanek** [comment]
The current code doesn't appear to have this problem:
https://github.com/meeting-room-booking-system/mrbs-code/blob/main/web/lib/MRBS/DB_pgsql.php#L292-L303

MRBS 1.11.0 was released 10 months ago, you'd be better off with MRBS 1.11.4 released last week.

**Comment by seven0f** [comment]
Terrific, I will try that.

**Comment by campbell-m** [comment]
> The current code doesn't appear to have this problem

Yes, it was fixed in [5f3a2b7](https://github.com/meeting-room-booking-system/mrbs-code/commit/5f3a2b7c3949dc5eccc7da17fb0c5bb99243de39).  See Issue #207.
